crates/net/network/src/transactions/mod.rs

```diff
@@ -1015,25 +1015,46 @@ where
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let this = self.get_mut();
 
+        // If the budget is exhausted we manually yield back control to tokio. See
+        // `NetworkManager` for more context on the design pattern.
+        let budget = &mut 1024;
+        macro_rules! budget_checkpoint {
+            () => {
+                *budget -= 1;
+                if *budget == 0 {
+                    // make sure we're woken up again
+                    return Poll::Ready(())
```

Comment:
is there a reason why this isn't calling `wake_by_ref` like in `NetworkManager`?
```rust
                // make sure we're woken up again
                cx.waker().wake_by_ref();
```

Reply:
well it's the same thing technically :)

Reply:
ah no, it's not a stream!